pkg/proc/gdbserial/undo.go

```diff
@@ -158,3 +160,107 @@ func UndoIsRecording(recordingFile string) (result bool, err error) {
 
 	return bytes.Equal(marker, data), nil
 }
+
+// Fetch the output of a udbserver get_info command, split on ; and , characters.
+//
+// This is not (currently) implementing a proper parse of the data returned, just making it more
+// convenient to search.
+func undoGetInfo(conn gdbConn) []string {
+	info, err := conn.undoCmd("get_info")
+	if err != nil {
+		panic("Failed to retrieve Undo stop info.")
+	}
+	splitter := func(c rune) bool {
+		return c == ';' || c == ','
+	}
+	return strings.FieldsFunc(info, splitter)
+}
+
+func undoGetLogExtent(p *gdbProcess) (uint64, uint64) {
```

Comment:
It would be nice to have a comment documenting what this does. Also there is an inconsistency in the arguments for `undoGetInfo` (which takes a `gdbConn`) and `undoGetLogExtent` (which takes a `gdbProcess`) -- consider picking one approach and using it consistently if possible.

Reply:
Mmmm.  I'm inclined to have this layer be concerned with the `gdbProcess` consistently, since that represents (as far as I can see!) the overall debug session on this process - will do that.

Reply:
This can't stay at the `gdbProcess` level because, for error handling to work right, we need to call into some of these new functions from the `gdbConn` level.  I've standardised on that instead and I think it's fine.

Reply:
Resolved to use `gdbConn` everywhere - `gdbProcess` didn't fit once error handling was taken into account.